js/adapt-contrib-trickle.js

```diff
@@ -1,134 +1,472 @@
+    }
 
-      fromModel.set('_isTrickleAutoScrollComplete', true);
+    set isFinished(value) {
+      if (!Adapt.parentView) return;
+      Adapt.parentView.model.set('_isTrickleFinished', value);
+      if (value) {
+        Adapt.trigger('trickle:finished');
+        this.trigger('finished');
+      }
+    }
 
-      var trickle = Adapt.trickle.getModelConfig(fromModel);
-      var isAutoScrollOff = !trickle._autoScroll;
-      var hasTrickleButton = trickle._button._isEnabled;
+    /**
+     * Returns true if the current page was locked by trickle and is now killed
+     */
+    get isKilled() {
+      if (!Adapt.parentView) return false;
+      return !Adapt.trickle.isStarted || Adapt.parentView.model.get('_isTrickleKilled');
+    }
+
+    set isKilled(value) {
+      if (!Adapt.parentView) return;
+      Adapt.parentView.model.set('_isTrickleKilled', value);
+      if (value) {
+        Adapt.trigger('trickle:killed');
+        this.trigger('killed');
+      }
+    }
+
+    checkIfIsTrickleFinished() {
+      if (!Adapt.trickle.isStarted) return;
+      if (Adapt.trickle.isKilled || Adapt.trickle.isFinished) return;
+      if (Adapt.trickle.isTrickling) return;
+      Adapt.trickle.isFinished = true;
+    }
+
+    /**
+     * When each child view is added to its parent, check to see if trickle should
+     * stop the render at this child according to the locking state
+     * @param {ChildEvent} event
+     */
+    onAddChild(event) {
+      if (Adapt.trickle.isKilled) return;
+      if (event.hasRequestChild) {
+        this.applyLocks();
+      }
+      const isManagedByTrickleAndLocked = (event.model.get('_isTrickled') && event.model.get('_isLocked'));
+      if (!isManagedByTrickleAndLocked) return;
+      event.stop();
+    }
+
+    /**
+     * Make the framework try to render the next set of children
+     */
+    async continue() {
+      this.applyLocks();
+      await Adapt.parentView.addChildren();
+      await Adapt.parentView.whenReady();
+    }
+
+    /**
+     * Scroll to the next section as defined by the trickle configuration on the
+     * specified model
+     * @param {Backbone.Model} fromModel
+     */
+    async scroll(fromModel) {
+      const trickle = this.getModelConfig(fromModel);
+      if (!trickle || !trickle._isEnabled) return false;
+
+      const isArticleWithOnChildren = (fromModel.get('_type') === 'article' && trickle._onChildren);
+      if (isArticleWithOnChildren) return false;
+
+      const isAutoScrollOff = !trickle._autoScroll;
+      const hasTrickleButton = trickle._button._isEnabled;
       if (isAutoScrollOff && !hasTrickleButton) {
         return;
       }
 
-      var scrollTo = trickle._scrollTo;
-      if (scrollTo === undefined) scrollTo = '@block +1';
-
-      var scrollToId = '';
-      switch (scrollTo.substr(0,1)) {
+      const scrollTo = trickle._scrollTo;
+      const firstCharacter = scrollTo.substr(0, 1);
+      let scrollToId = '';
```

Comment:
same with this empty string - it could be set in the initialize function

Reply:
Why only the empty string and not any of the other variables in this function?

Reply:
ignore me, I've thought about it a little more and its fine - I think what you've done is best.